### PR TITLE
Add 'min-acl' to let users determine the minimum access level to gene…

### PR DESCRIPTION
**Pull request by minuscorp**
…rate the documentation

#### Breaking
- None

#### Enhancements
- Adds `min-acl` parameter that allows `private | fileprivate | internal | public | open` options, defaults to `public` as for now.

#### Bug Fixes
- None

### CODE REVIEW

**Comment by minuscorp** [comment]
For some reason, `sourcedocs generate --min-acl "private" --input-folder Ellu --output-folder Docs -- -workspace Ellu.xcworkspace -scheme Ellu CODE_SIGNING_ALLOWED="NO"` does not detect all files in the project, as the command `sourcekitten doc -- -workspace Ellu.xcworkspace -scheme Ellu` does. Some insight on this?

**Comment by minuscorp** [comment]
So I have tested with a SPM of my own and it works, but I cannot make it work with a Xcode project, sourcekitten seems to return and empty list of source files, while the direct `sourcekitten` command works, weird

**Comment by minuscorp** [comment]
So in the end was a problem with SourceKitten version, updated it, working like a charm. Ready to review

**Comment by minuscorp** [comment]
Not sure why certain steps are failing on the CI (are passing fine on local)

File: Sources/SourceDocs/Classes/SourceDocs.swift [hunk 1]
```diff
@@ -26,5 +26,74 @@ struct SourceDocs {
...
+        return lhs.priority < rhs.priority
+    }
+    
+    static func > (lhs: AccessLevel, rhs: AccessLevel) -> Bool {
+        return lhs.priority > rhs.priority
+    }
+    
+    static func == (lhs: AccessLevel, rhs: AccessLevel) -> Bool {
+        return lhs.priority == rhs.priority
+    }
```
Comment (eneko) [inline Sources/SourceDocs/Classes/SourceDocs.swift]: This is pretty cool, nice idea!

**Comment by minuscorp** [comment]
As far as I've tested, this works fine on SPM modules and Xcode projects (the failure was because the outdated Sourcekitten dependency, which ironically makes the build to fail on the CI but not on local 🤷‍♂ )

**Comment by minuscorp** [comment]
I think that the issue with the CI can be the usage of a pretty old Xcode image, I world try to bump it at least to Xcode 10, that has support for Swift 5

**Comment by minuscorp** [comment]
Updated the Travis workflow, minimum toolchain needed is Swift 5.0 (SourceKitten)

File: Sources/SourceDocs/Classes/SwiftDocDictionary.swift [hunk 2]
```diff
@@ -16,8 +16,8 @@ extension Dictionary where Key == String, Value == Any {
         return self[key.rawValue] as? T
     }
 
-    var hasPublicACL: Bool {
-        return self["key.accessibility"] as? String == "source.lang.swift.accessibility.public"
+    var ACL: AccessLevel {
+        return AccessLevel(rawValue: self["key.accessibility"] as? String ?? "source.lang.swift.accessibility.public")!
```
Comment (eneko) [inline Sources/SourceDocs/Classes/SwiftDocDictionary.swift]: Please rename `ACL` to `accessLevelControl`, `accessLevel`, or `acl`. Properties shouldn't be uppercased.

In addition, please remove the forced-unwrap, and default to `private` ACL if no access level is available.

Comment (eneko) [reply Sources/SourceDocs/Classes/SwiftDocDictionary.swift]: Actually, I'll take care of these once landed, thank you!

**Comment by minuscorp** [comment]
Have a spare week, want me to dive to finish this? @eneko

**Pull request merged**